src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -244,7 +244,8 @@ async Task Process(BasicDeliverEventArgs message)
                 {
                     try
                     {
-                        var messageContext = new MessageContext(messageId, headers, message.Body ?? new byte[0], transportTransaction, tokenSource, contextBag);
+                        var msgHeaders = new Dictionary<string, string>(headers);
```

Comment:
This introduces an unneeded dictionary copy/allocations on the main message processing path. Instead consider creating a new header dictionary by calling `messageConverter.RetrieveHeaders(message)` only when an error has actually occurred and potential modifications to the original collection could be observed.

Reply:
This was just a basic test to see whether it passes the tests, I'll look into more efficient approaches. I wanted to commit these changes because this exact change caused the test to fail locally due to weird exceptions: `RabbitMQ.Client.Exceptions.BrokerUnreachableException : None of the specified endpoints were reachable` and I wanted to see whether it happens on the build server too.